switches/http/http.cpp

```diff
@@ -0,0 +1,1280 @@
+    {
+        do
+        {
+            sendAvailableResponses();
+            step(2);
+            if (verbose && streamsToShutdown.size() > 0)
+            {
+                cerr << "HttpInterface::run "<< streamsToShutdown.size() <<" streams to shut down";
+                for (std::set<Dashel::Stream*>::iterator si = streamsToShutdown.begin(); si != streamsToShutdown.end(); si++)
+                    cerr << " " << *si;
+                cerr << endl;
+            }
+            if (!streamsToShutdown.empty())
+            {
+                std::set<Dashel::Stream*>::iterator i = streamsToShutdown.begin();
+                Dashel::Stream* stream_to_shutdown = *i;
+                streamsToShutdown.erase(*i); // invalidates iterator
+                try
+                {
+                    if (verbose)
+                        cerr << stream_to_shutdown << " shutting down stream" << endl;
+                    shutdownStream(stream_to_shutdown);
+                }
+                catch(Dashel::DashelException& e)
+                { }
+            }
+        } while (iterations-- != 0 and asebaStream != 0);
+        for (StreamResponseQueueMap::iterator i = pendingResponses.begin(); i != pendingResponses.end(); i++)
+            unscheduleAllResponses(i->first);
+    }
+    
+    void HttpInterface::connectionCreated(Dashel::Stream *stream)
+    {
+        if (!asebaStream)
+        {
+            // this is the connection to the Thymio-II
+            std::cout << "Incoming Aseba connection from " << stream->getTargetName() << endl;
+            asebaStream = stream;
+        }
+        else
+        {
+            // this is an incoming HTTP connection
+            if (verbose)
+                cerr << stream << " Connection created to " << stream->getTargetName() << endl;
+            
+            assert( pendingResponses[stream].empty() );
+        }
+    }
+    
+    void HttpInterface::connectionClosed(Stream * stream, bool abnormal)
+    {
+        if (stream == asebaStream)
+        {
+            // first close all HTTP connections
+            for (StreamResponseQueueMap::iterator m = pendingResponses.begin(); m != pendingResponses.end(); m++)
+                closeStream(m->first);
+            // then stop the hub
+            asebaStream = 0;
+            if (verbose)
+                cerr << "Connection closed to Aseba target" << endl;
+            stop();
+        }
+        else
+        {
+            if (verbose)
+                cerr << stream << " Connection closed to " << stream->getTargetName() << endl;
+            unscheduleAllResponses(stream);
+            pendingResponses.erase(stream);
+            unsigned num = streamsToShutdown.erase(stream);
+            if (verbose)
+                cerr << stream << " Connection closed, removed " << num << " pending shutdowns" << endl;
+        }
+    }
+    
+    void HttpInterface::nodeDescriptionReceived(unsigned nodeId)
+    {
+        if (verbose)
```

Comment:
I meant in the following lines, there are 1 and 2 space indentation, while in the rest of the code it is 4 spaces.